src/lexer.c

```diff
@@ -403,14 +433,83 @@ void tryParseExportStatement () {
 
 char16_t readExportAs (char16_t* startPos, char16_t* endPos) {
```

Comment:
The `readExportAs` scan is specifically designed to always be at the possible start position of `as ...`. As per the previous comment this invariant is being changed which opens up a class of bugs. We should make sure to keep this invariant.

Reply:
OK, I will consider recover it and use new process.